bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/graphics/ImageData.java

```diff
@@ -400,6 +456,59 @@ public ImageData(String filename) {
 		i.delayTime);
 }
 
+/**
+ * Constructs an <code>ImageData</code> loaded from a file with the
+ * specified name. Throws an error if an error occurs loading the
+ * image, or if the image has an unsupported type.
+ *
+ * @param filename the name of the file to load the image from (must not be null)
+ * @param zoom the zoom factor to apply when rasterizing a SVG.
+ * A value of 0 means that the standard method for loading should be used.
+ * This case is equivalent to calling {@link ImageLoader#load(String)}.
+ *
+ * A value above 0 specifies a scaling factor for the output image. For example:
+ * <ul>
+ *   <li>A value of 100 maintains the original size of the SVG when rasterized.</li>
+ *   <li>A value of 200 doubles the size of the rasterized image.</li>
+ *   <li>A value of 50 reduces the size of the rasterized image to half.</li>
+ * </ul>
+ * The scaling is applied uniformly to both width and height.
+ *
+ * @exception IllegalArgumentException <ul>
+ *    <li>ERROR_NULL_ARGUMENT - if the file name is null</li>
+ * </ul>
+ * @exception SWTException <ul>
+ *    <li>ERROR_IO - if an IO error occurs while reading from the file</li>
+ *    <li>ERROR_INVALID_IMAGE - if the image file contains invalid data</li>
+ *    <li>ERROR_UNSUPPORTED_FORMAT - if the image file contains an unrecognized format</li>
+ * </ul>
+ *
+ * @since 3.129
+ */
+public ImageData(String filename, int zoom) {
```

Comment:
In general using String to model file-system paths is a bad idea. But that's how it's done in the existing API.
I have created a PR with the proposal to replace it:
- https://github.com/eclipse-platform/eclipse.platform.swt/pull/1767

If that is successful I think we should use `File` directly in new methods.